opm/simulators/wells/MultisegmentWellEval.cpp

```diff
@@ -164,6 +164,15 @@ getWellConvergence(const WellState& well_state,
                                   report,
                                   deferred_logger);
 
+    // checking the flow direction
+    const double sign = baseif_.isProducer() ? -1. : 1.;
+    const auto weight_total_flux = this->primary_variables_.getWQTotal() * sign;
+    constexpr int dummy_phase = -1;
+    // TODO: maybe a small negative value if needed for robustness
```

Comment:
Suggest to skip this test (also in StandardWellEval.cpp) if the well is STOPPED or has a zero rate control. Then we should not need to add an epsilon here.

Reply:
zero rate control is a little troublesome to detect. What I can do here is that we only check BHP and THP controlled wells.